### PR TITLE
Support boolean attributes

### PR DESCRIPTION
**Pull request by stephencelis**

### CODE REVIEW

File: Sources/HTML/Attributes.swift [hunk 1]
```diff
@@ -1,6 +1,29 @@
+public let action = AttributeKey<String>(key: "action")
```
Comment (stephencelis) [inline Sources/HTML/Attributes.swift]: Added a bunch of these. More to come!

File: Sources/HTML/HTML.swift [hunk 2]
```diff
@@ -21,8 +21,8 @@ public func text(_ string: String) -> Node {
   return .text(string)
 }
 
-public protocol IsValue {
-  func toValue() -> String
+public protocol Value {
+  func render(with key: String) -> String?
```
Comment (stephencelis) [inline Sources/HTML/HTML.swift]: This change was necessary to support booleans, though the API doesn't feel quite right yet.

File: Sources/HTML/HTML.swift [hunk 3]
```diff
@@ -49,29 +45,34 @@ public func node(_ name: String, _ content: [Node]) -> Node {
   return .element(.init(name: name, attribs: [], content: content))
 }
 
-extension String: IsValue {
-  public func toValue() -> String {
-    return self
+extension String: Value {
+  public func render(with key: String) -> String? {
+    return "\(key)=\"\(self)\""
```
Comment (stephencelis) [inline Sources/HTML/HTML.swift]: This is where the API gets a bit annoying and repetitive per `Value`. Wanna come up with a more generalizable way of quoting attributes.

File: Sources/HTML/HTML.swift [hunk 4]
```diff
@@ -21,8 +21,8 @@ public func text(_ string: String) -> Node {
   return .text(string)
 }
 
-public protocol IsValue {
-  func toValue() -> String
+public protocol Value {
```
Comment (stephencelis) [inline Sources/HTML/HTML.swift]: A lil rename. Is this OK?

Comment (mbrandonw) [inline Sources/HTML/HTML.swift]: yeah def

**Pull request merged**